src/combinations.rs

```diff
@@ -203,15 +249,39 @@ where
     }
 }
 
-impl<I> FusedIterator for Combinations<I>
+impl<I, Idx> FusedIterator for CombinationsGeneric<I, Idx>
 where
     I: Iterator,
     I::Item: Clone,
+    Idx: PoolIndex<I::Item>,
 {
 }
 
+impl<I: Iterator> Combinations<I> {
+    /// Resets this `Combinations` back to an initial state for combinations of length
+    /// `k` over the same pool data source. If `k` is larger than the current length
+    /// of the data pool an attempt is made to prefill the pool so that it holds `k`
+    /// elements.
```

Comment:
Thanks for correcting the comment.

Reply:
I didn't actually touch this comment, the function has just moved to a separate `impl` since it is only used by powerset and so only makes sense to have for variable length combinations.